web/blog/zeek-in-the-shell/index.md

```diff
@@ -0,0 +1,179 @@
+  `eve-log.filename`. Setting `/dev/stdout` as filename makes Suricata write to
+  stdout.
+- We must set `logging.outputs.0.console.enabled` to `no` because Suricata
+  doesn't write startup log messages on stderr. Since they are no valid JSON, we
+  would otherwise create an invalid JSON output stream.
+
+## User-defined Operators
+
+Now that we have both Zeek and Suricata at our fingertips, how can we work with
+their output more easily? This is where Tenzir comes into play-easy
+[pipelines](/docs/language/pipelines) for security teams to acquire,
+[reshape](/docs/user-guides/get-started), and route event data.
+
+For our use case we look at the [`shell`](/docs/operators/shell) operator in
+more detail: it lifts any Unix command line into a pipeline. Here are two
+examples that rely on our two shell scripts:
+
+```bash
+# Zeek
+zcat pcap.gz | zeekify | tenzir \
+  'read zeek-json
+   | where #type == "zeek.conn"
+   | summarize n=count_distinct(id.orig_h) by id.resp_h
+   | sort n desc'
+# Suricata
+zcat pcap.gz | suricatify | tenzir \
+  'read suricata
+   | where #type == "suricata.flow"
+   | summarize n=count_distinct(src_ip) by dst_ip
+   | sort n desc'
+```
+
+The two aggregations actually do the same thing: count the number of unique
+source IP addresses per destination IP address.
+
+It's a bit unwiedly to write such a command line that requires an external shell
+script to work. This where [user-defined operators](/operators/user-defined)
+come into play. You can write a custom `zeek` and `suricata` operator and ditch
+the shell script:
+
+```yaml title="tenzir.yaml"
+tenzir:
+  operators:
+    zeek: >
+     shell "zeek -r - LogAscii::output_to_stdout=T \
+            JSONStreaming::disable_default_logs=T \
+            JSONStreaming::enable_log_rotation=F \
+            json-streaming-logs"
+     | parse zeek-json
+    suricata: >
+     shell "suricata -r /dev/stdin \
+            --set outputs.1.eve-log.filename=/dev/stdout \
+            --set logging.outputs.0.console.enabled=no"
+     | parse suricata
+```
+
+The difference stands out when you look now at the pipeline definition:
+
+```text title=Zeek
+zeek
+| where #type == "zeek.conn"
+| summarize n=count_distinct(id.orig_h) by id.resp_h
+| sort n desc
+```
+
+```text bash title=Suricata
+suricata
+| where #type == "suricata.flow"
+| summarize n=count_distinct(src_ip) by dst_ip
+| sort n desc
+```
+
+It's pretty convenient to drop packets into a Tenzir pipeline, process them with
+our favorite tools, and then perform fast in-situ analytics on them. The nice
+thing is that operators compose: a new operator automatically works with all
+existing ones.
+
+## How does it work?
+
+Under the hood, the `shell` operator forks the `tenzir` process and spawns
+`zeek` or `suricata` as child process. The operator then rewires stdin and
+stdout of the `tenzir` process such their byte streams get routed through stdin
+and stdout of the forked child.
```

Comment:
And similar to one of the previous comments: I think the flow could be improved by either moving this after the "standard approach", or by moving the standard approach after the `shell` approach.